isolate_example/lib/page_three.dart

```diff
@@ -11,14 +11,243 @@
+
+    List<int> randNums = [];
+    for (int i = 0; i < 100; i++) {
+      randNums.clear();
+
+      for (int j = 0; j < 1000000; j++) {
+        randNums.add(rng.nextInt(100));
+      }
+      await sendNumbers(randNums);
+    }
+  }
+
+  Future<void> sendNumbers(List<int> numList) {
+    return Future<void>(() {
+      _newIceSP.send(numList);
+    });
+  }
+
+  Isolate get newIsolate => _newIsolate;
+
+  void dispose() {
+    super.dispose();
+    _newIsolate?.kill(priority: Isolate.immediate);
+    _newIsolate = null;
+  }
+}
+
+class RunningList extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    return Center(
-      child: Text('Welcome to the Data Transfer Page'),
+    final controller = Provider.of<DataTransferIsolateController>(context);
+    List<String> progress = controller.currentProgress;
+
+    return DecoratedBox(
+      decoration: BoxDecoration(
+        color: Colors.grey[200],
+      ),
+      child: new ListView.builder(
+        itemCount: progress.length,
+        itemBuilder: (context, index) {
+          return Column(
+            children: [
+              Card(
+                child: ListTile(
+                  title: Text(progress[index]),
+                ),
+                color: Colors.lightGreenAccent,
+              ),
+              Divider(
+                color: Colors.blue,
+                height: 3,
+              ),
+            ],
+          );
+        },
+      ),
     );
   }
 }
+
+Future<void> _secondIsolateEntryPoint(SendPort callerSP) async {
+  List<int> randNums = [];
+  ReceivePort newIceRP = ReceivePort();
+  callerSP.send(newIceRP.sendPort);
+
+  newIceRP.listen(
+    (message) {
+      if (message is String && message == 'Start') {
+        int sum = 0;
+        Random rng = new Random();
+        for (int i = 0; i < 100000000; i++) {
+          sum += rng.nextInt(100);
+
+          if ((i + 1) % 1000000 == 0) {
+            callerSP.send((i + 1) ~/ 1000000);
+          }
+        }
+
+        callerSP.send('done');
+        return sum;
+      } else {
+        randNums.addAll(message as List<int>);
+
+        callerSP.send(randNums.length ~/ 1000000);
+
+        if (randNums.length == 100000000) {
+          callerSP.send('transfer done');
+          addUp(randNums);
+          callerSP.send('done');
+          randNums.clear();
+        }
+      }
+    },
+  );
+}
+
+int addUp(List<int> numList) {
+  int sum = 0;
+//  for (int i = 0; i < numList.length; i++) {
+  for (int i in numList) {
+    sum += i;
+  }
+  return sum;
+}
+
+Widget newButtons(context) {
+  final controller =
+      Provider.of<DataTransferIsolateController>(context, listen: false);
+  return ButtonBar(
+    alignment: MainAxisAlignment.center,
+    children: [
+      Column(
+        children: [
+          RaisedButton(
+            child: const Text('Transfer Data to 2nd Isolate'),
+            elevation: 8.0,
+            onPressed: () => controller.generateRandomNumbers(),
```

Comment:
Can you use a tear-off here and below, rather than defining a new function?

```
onPressed: controller.generateRandomNumbers,
```

https://dart.dev/guides/language/effective-dart/usage#dont-create-a-lambda-when-a-tear-off-will-do